z.mlp

```diff
@@ -124,13 +124,24 @@ let erem a b =
    if sign r >= 0 then r else add r (abs b)
 
 let gcdext u v =
-  let g,s,z = gcdext_intern u v in
-  if z then g, s, div (sub g (mul u s)) v
-  else g, div (sub g (mul v s)) u, s
+  match sign u, sign v with
+  (* special cases: one argument is null *)
```

Comment:
I don't fully understand what `gcdext` does, so I didn't review these special cases, but I trust you.

Reply:
As far as I understand, gcdext(a,b) implements Bezout's theorem: in addition to gcd(a,b), it returns u and v such that a*u + b*v = gcd(a,b). However, which u and v are precisely chosen by gmp is a bit complicated and seems to have changed in the past. So, I don't want to make any additional promise. It is not difficult though to find u and v that work in the case where a or b is 0 (basically choosing in -1, 0, 1).

Reply:
All right.  The documentation of `gcdext` in z.mlip is quite clear about `u` and `v` being underspecified.